share/getters/shrex.go

```diff
@@ -92,6 +94,9 @@ func (sg *ShrexGetter) GetEDS(ctx context.Context, root *share.Root) (*rsmt2d.Ex
 		case errors.Is(getErr, p2p.ErrNotFound):
 			getErr = share.ErrNotFound
 			setStatus(peers.ResultCooldownPeer)
+		case errors.Is(getErr, routing.ErrNotFound):
+			// peer is no longer reachable, remove it from all pools
+			setStatus(peers.ResultRemovePeer)
```

Comment:
Ah, actually, this won't happen after we update to [0.27.0](https://github.com/libp2p/go-libp2p/releases/tag/v0.27.0), because of https://github.com/libp2p/go-libp2p/pull/2169

Reply:
Made an update https://github.com/celestiaorg/celestia-node/pull/2127

Reply:
What error would be returned in case peer is disconnected?

Reply:
Errors are not returned on disconnect, but they will be returned on subsequent reconnect if it fails

Reply:
Usually it's something like dial backoff

Reply:
@walldiss, https://github.com/libp2p/go-libp2p/blob/master/p2p/net/swarm/swarm_dial.go#L37-L49

It's either ErrDialBackoff or ErrAllDialsFailed